### PR TITLE
Smaller bundle size

### PR DESCRIPTION
**Pull request by airhorns**
This slims down the base bundle size of a generated client by about 22% uncompressed and 28% compressed woop woop! See the commits for individual changes / size reductions.

Before:

```
vite v4.4.7 building for production...
✓ 223 modules transformed.
dist/api-read-stats.html      230.59 kB │ gzip: 48.89 kB
dist/test-bundle-api-read.js  253.91 kB │ gzip: 47.33 kB
dist/api-read-stats.html       231.12 kB │ gzip: 48.95 kB
dist/test-bundle-api-read.cjs  168.40 kB │ gzip: 39.39 kB
✓ built in 965ms
vite v4.4.7 building for production...
✓ 499 modules transformed.
dist/react-read-stats.html      321.86 kB │ gzip: 59.00 kB
dist/test-bundle-react-read.js  359.09 kB │ gzip: 76.48 kB
dist/react-read-stats.html       324.49 kB │ gzip: 59.05 kB
dist/test-bundle-react-read.cjs  243.18 kB │ gzip: 64.60 kB
✓ built in 1.34s
vite v4.4.7 building for production...
✓ 1033 modules transformed.
dist/shopify-read-stats.html      534.09 kB │ gzip:  79.63 kB
dist/test-bundle-shopify-read.js  678.46 kB │ gzip: 135.16 kB
dist/shopify-read-stats.html       534.63 kB │ gzip:  79.70 kB
dist/test-bundle-shopify-read.cjs  477.27 kB │ gzip: 115.70 kB
✓ built in 2.50s
```

After:

```
vite v4.4.7 building for production...
✓ 173 modules transformed.
dist/api-read-stats.html           210.92 kB │ gzip: 46.91 kB
dist/browser-ponyfill-a950d05c.js   12.53 kB │ gzip:  3.50 kB
dist/test-bundle-api-read.js       199.78 kB │ gzip: 34.45 kB
dist/api-read-stats.html            211.51 kB │ gzip: 46.98 kB
dist/browser-ponyfill-0b1c63f2.cjs    8.88 kB │ gzip:  3.07 kB
dist/test-bundle-api-read.cjs       130.06 kB │ gzip: 28.07 kB
✓ built in 774ms
vite v4.4.7 building for production...
✓ 449 modules transformed.
dist/react-read-stats.html         302.31 kB │ gzip: 56.84 kB
dist/test-bundle-react-read.js       0.09 kB │ gzip:  0.10 kB
dist/browser-ponyfill-aae43e35.js   12.44 kB │ gzip:  3.50 kB
dist/react-read-b8f94e10.js        305.18 kB │ gzip: 63.69 kB
dist/react-read-stats.html          306.31 kB │ gzip: 56.96 kB
dist/test-bundle-react-read.cjs       0.17 kB │ gzip:  0.16 kB
dist/browser-ponyfill-1992ee4c.cjs    8.83 kB │ gzip:  3.08 kB
dist/react-read-e3b5cb04.cjs        204.97 kB │ gzip: 53.20 kB
✓ built in 1.16s
vite v4.4.7 building for production...
✓ 983 modules transformed.
dist/shopify-read-stats.html       514.75 kB │ gzip:  77.59 kB
dist/test-bundle-shopify-read.js     0.12 kB │ gzip:   0.12 kB
dist/browser-ponyfill-d3f0b39a.js   12.45 kB │ gzip:   3.50 kB
dist/shopify-read-62230a97.js      624.51 kB │ gzip: 122.38 kB
dist/shopify-read-stats.html        515.26 kB │ gzip:  77.60 kB
dist/test-bundle-shopify-read.cjs     0.19 kB │ gzip:   0.17 kB
dist/browser-ponyfill-8417ec9c.cjs    8.83 kB │ gzip:   3.08 kB
dist/shopify-read-25c98907.cjs      439.00 kB │ gzip: 104.22 kB
```

Here's the bundle analyzer for the simplest test bundle at the end: https://bundle-sizes--development.gadget.app/sept-19-2023/api-read-stats.html

### CODE REVIEW

File: packages/api-client-core/src/GadgetConnection.ts [hunk 1]
```diff
@@ -103,7 +97,10 @@ export class GadgetConnection {
     } else if (typeof window != "undefined" && window.fetch) {
       this._fetchImplementation = window.fetch.bind(window);
     } else {
-      this._fetchImplementation = fetchPolyfill;
+      this._fetchImplementation = async (...args: [any]) => {
```
Comment (scott-rc) [inline packages/api-client-core/src/GadgetConnection.ts:100]: nit

```suggestion
      this._fetchImplementation = async (...args: any[]) => {
```

**Review by scott-rc — APPROVED** [review]
LGTM 👍

File: packages/api-client-core/src/GadgetConnection.ts (hunk 1 above)
Comment (airhorns) [reply packages/api-client-core/src/GadgetConnection.ts:100]: Fun fact: typescript complains if you don't use a tuple type for the args, that's why its like this

Comment (scott-rc) [reply packages/api-client-core/src/GadgetConnection.ts:100]: TIL 🧠

**Review by thegedge — APPROVED** [review]
🚀

**Pull request merged**